base/deprecated.jl

```diff
@@ -1033,4 +1034,25 @@ end))
 @deprecate_binding cycle     Iterators.cycle
 @deprecate_binding repeated  Iterators.repeated
 
+# Not exported, but probably better to have deprecations anyway
+function reduced_dims(::Tuple{}, d::Int)
+    d < 1 && throw(ArgumentError("dimension must be ≥ 1, got $d"))
+    ()
+end
+reduced_dims(::Tuple{}, region) = ()
+function reduced_dims(dims::Dims, region)
+    Base.depwarn("`reduced_dims` is deprecated for Dims-tuples; pass `indices` instead", :reduced_dims)
+    map(r->last(r), reduced_dims(map(n->OneTo(n), dims), region))
```

Comment:
`map(last, reduce_dims(map(OneTo, dims), region))`? Shouldn't be necessary to construct anonymous functions in those places, I would think.